File: backend/questao_2.py
def q2_contar_frequencia_palavra(text):
    """Realiza uma contagem de quantas vezes uma determinada palavra ocorre
    Ao final, preferencialmente, ordenar o dicionário pelo volume de ocorrência

    Saída esperada: {'hello': 2, 'world': 1}
    """

    dict = {}
    text = text.lower()
    text = text.split()
    for word in text:
        if word in dict:
            dict[word] += 1
        else:
            dict[word] = 1
        
    return dict

print(q2_contar_frequencia_palavra("Hello world hello"))

File: frontend/questao_2.js
function q2ContarFrequenciaPalavra(text) {
    const dict = {};
    
    text = text.toLowerCase().split(' ');
    
    for (const word of text) {
        if (dict[word]) {
            dict[word] += 1;
        } else {
            dict[word] = 1;
        }
    }
    
    
    return Object.fromEntries(
        Object.entries(dict).sort(([,a], [,b]) => b - a)
    );
}

console.log(q2ContarFrequenciaPalavra("Hello world hello"));
